zebra-chain/src/parameters/network.rs

```diff
@@ -108,17 +163,69 @@ impl fmt::Display for Network {
 }
 
 impl Network {
+    /// Creates a new [`Network::Testnet`] with the default Testnet [`NetworkParameters`].
+    pub fn new_default_testnet() -> Self {
+        Self::Testnet(Arc::new(NetworkParameters::default()))
+    }
+
+    /// Creates a new configured [`Network::Testnet`] with the provided Testnet [`NetworkParameters`].
+    pub fn new_configured_testnet(params: NetworkParameters) -> Self {
+        Self::Testnet(Arc::new(params))
+    }
+
+    /// Returns true if the network is the default Testnet, or false otherwise.
+    pub fn is_default_testnet(&self) -> bool {
+        if let Self::Testnet(params) = self {
+            params.is_default_testnet()
+        } else {
+            false
+        }
+    }
+
+    /// Returns the [`NetworkKind`] for this network.
+    pub fn kind(&self) -> NetworkKind {
+        match self {
+            Network::Mainnet => NetworkKind::Mainnet,
+            Network::Testnet(_) => NetworkKind::Testnet,
+        }
+    }
+
+    /// Returns the default [`Network`] for a [`NetworkKind`].
+    pub fn from_kind(kind: NetworkKind) -> Self {
+        match kind {
+            NetworkKind::Mainnet => Self::Mainnet,
+            NetworkKind::Testnet => Self::new_default_testnet(),
+        }
+    }
+
     /// Returns an iterator over [`Network`] variants.
     pub fn iter() -> impl Iterator<Item = Self> {
         // TODO: Use default values of `Testnet` variant when adding fields for #7845.
-        [Self::Mainnet, Self::Testnet].into_iter()
+        [Self::Mainnet, Self::new_default_testnet()].into_iter()
+    }
+
+    /// Returns true if the maximum block time rule is active for `network` and `height`.
+    ///
+    /// Always returns true if `network` is the Mainnet.
+    /// If `network` is the Testnet, the `height` should be at least
+    /// TESTNET_MAX_TIME_START_HEIGHT to return true.
+    /// Returns false otherwise.
+    ///
+    /// Part of the consensus rules at <https://zips.z.cash/protocol/protocol.pdf#blockheader>
+    pub fn is_max_block_time_enforced(&self, height: block::Height) -> bool {
+        match self {
+            Network::Mainnet => true,
+            // TODO: Move `TESTNET_MAX_TIME_START_HEIGHT` to a field on NetworkParameters (#8364)
+            Network::Testnet(_params) => height >= super::TESTNET_MAX_TIME_START_HEIGHT,
```

Comment:
Consider adding a test that checks that this returns the correct value for the default testnet, either in this PR or the one that moves this to a field on `NetworkParameters`.